src/timeoffset.rs

```diff
@@ -0,0 +1,362 @@
+use crate::{Epoch, TimeScale, Unit};
+
+/// Errors during [TimeOffset] exploitation
+#[derive(Debug)]
+pub enum TimeOffsetError {
+    /// Timescales should not be identical when defining a new [TimeOffset]!
+    IdenticalTimescales,
+    /// Epoch is not expressed in one of the two supported [TimeScale]s!
+    NotSupportedTimescale,
+    /// Epoch should be expressed in the left-hand side [TimeScale].
+    InvalidTimescale,
+    /// [TimeOffset] is outdated and should not apply (difference is too large): should be updated weekly at least!
+    OutdatedTimeOffset,
+}
+
+/// [TimeOffset] used in [TimeShift]ing operations
```

Comment:
I think this documentation should provide an example of when this is useful, like in your PR description with the UTC-GPST difference.

The documentation references the `TimeShift` struct, but that isn't in the PR.

Reply:
About the TimeShift stuff, sorry I did not intend to join it to this PR. It's part of my own stuff. I think it's up to the user to do whatever they have, while Hifitime remains the building blocs